api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -63,16 +64,43 @@ public Obs getObsByUuid(String uuid) {
 		
 		handleEncounterReference("e", encounterReference).ifPresent(c -> criteria.createAlias("encounter", "e").add(c));
 		handlePatientReference(criteria, patientReference, "person");
-//		handleBoolean("hasGroupMembers",hasMemberReference);
-
-		handleDateRange("valueDatetime",valueDateParam);
-		handleDateRange("obsDatetime",date);
+		handleHasMemberReference("hm", hasMemberReference).ifPresent(c -> criteria.createAlias("groupMembers", "hm").add(c));
```

Comment:
@ibacher Please have a look at the handlers. I am still left with the `has-member.code` parameter. Not very sure about how to convert value to codeable concept. I guess something like this 

```java
handleAndListParamBySystem(valueConcept, (system, tokens) -> {
CodeableConcept codeableConcept=new CodeableConcept();
		codeableConcept.addCoding().setCode(Not sure what will come here).setSystem(system);
});
```

Reply:
@ibacher Please have a look

Reply:
I actually don't think that you need to turn the `has-member.code` parameter into a codeable concept. Just refactor out the relevant part of the `handleCodedConcept` method from `FhirObservationDaoImpl` into `BaseDaoImpl` and call that.

Reply:
@ibacher I think that we'll have to handle it by doing something like this

```java
protected void handleHasMemberReference(Criteria criteria, ReferenceParam hasMemberReference) {
		if (hasMemberReference != null) {
			criteria.createAlias("obsGroup", "og");

			if (hasMemberReference.getChain() != null) {
				switch (hasMemberReference.getChain()) {
					case Observation.SP_CODE:
						criteria.add(eq("og.concept", hasMemberReference.getValue()));
						break;
					case "":
						criteria.add(eq("og.uuid", hasMemberReference.getValue()));
						break;
				}
			}
		}
	}
```

but I am not sure that should we use `obsGroup` or `groupMembers`

Reply:
@ibacher Isn't it better to have only one parameter as `hasMember` and through that we could search by `hasMember.code` as well. Will the above code not be an good option to use?

Reply:
So the property you should use is `groupMembers`. And, you're right that we'll need two branches here, but the handling for `SP_CODE` will need to be a bit more complex to handle:

`/ws/fhir2/Observation?has-member.code=http://loinc.org|8854-6`

😄

Reply:
@VaishSiddharth Do you see what I'm getting at here?

Reply:
@ibacher Yes you mean to say that the `code` part will be a bit difficult to handle, so no problem in that but can't we include that part in this handle function itself?

``` java 
protected void handleHasMemberReference(Criteria criteria, ReferenceParam hasMemberReference) {
		if (hasMemberReference != null) {
			criteria.createAlias("obsGroup", "og");

			if (hasMemberReference.getChain() != null) {
				switch (hasMemberReference.getChain()) {
					case Observation.SP_CODE:

//Call a function similar to handleValueCodedConcept
						criteria.add(eq("og.concept", hasMemberReference.getValue()));
						break;
					case "":
						criteria.add(eq("og.uuid", hasMemberReference.getValue()));
						break;
				}
			}
		}
	}
```

Reply:
Yes, that's the idea!